readme.md

```diff
@@ -109,6 +109,12 @@ Default: `false`
 
 Deep changes will not trigger the callback. Only changes to the immediate properties of the original object.
 
+##### clone
+
+Type: `function`
+
+A function that should return a clone of a value. Enables the "previous" value in the onChange callback when array methods are called.
```

Comment:
Are there any other use-cases for this other than for array? Otherwise, it would be nicer to just call it `previousValueForArray: true` or something and handle the cloning for the user.

Reply:
I agree that it would be nicer, I'm just concerned about bloating the code. To do a proper clone that "just works" we would more than double the size of this library (see: the most popular clone function on npm: https://github.com/pvorb/clone/blob/master/clone.js). A simpler clone function can be much smaller, but will not actually clone certain data types, or even blow up with things like circular references.  By providing this option, the user can select an appropriate clone function for their needs.

I suppose another option would be to return undefined in this case, and provide documentation that the user needs to do their own cloning prior to modifying the array?

Reply:
Ah, good point. I didn't think of that. I guess this is the best way then. Could maybe be useful to document why the user is expected to clone it themselves (the above reasons).